### PR TITLE
fix(onboarding): fixed broken "Log in with Keycard" flow and "Login" on desktop

### PR DESCRIPTION
**Pull request by saledjenic**
Corresponding `status-keycard-qt` PR:
- https://github.com/status-im/status-keycard-qt/pull/1

Fixes: #19866

### CODE REVIEW

File: ui/app/AppLayouts/Onboarding/pages/LoginScreen.qml [hunk 1]
```diff
@@ -222,7 +219,11 @@ OnboardingPage {
                                       root.keycardState === Onboarding.KeycardState.BlockedPUK
                 isKeycardEnabled: root.isKeycardEnabled
 
-                onSelectedProfileKeyIdChanged: {
+                onKeycardCreatedAccountChanged: {
+                    if (keycardCreatedAccount) {
```
Comment (alexjba) [inline ui/app/AppLayouts/Onboarding/pages/LoginScreen.qml]: This won't work unfortunately for some Android phones. Enabling the NFC too early (before qt finishes all the plumbing) will make nfc unusable.

That's the whole point of the `Scan keycard` button. We ensure the keycard scan is user-driven.

Comment (saledjenic) [reply ui/app/AppLayouts/Onboarding/pages/LoginScreen.qml]: Would like to see that in action, but I don't have an Android phone. Hope to get one in the next days and test.

Comment (alexjba) [reply ui/app/AppLayouts/Onboarding/pages/LoginScreen.qml]: > Would like to see that in action, but I don't have an Android phone. Hope to get one in the next days and test.

you can see it on IOS as well

Comment (saledjenic) [reply ui/app/AppLayouts/Onboarding/pages/LoginScreen.qml]: @alexjba I've tested these changes and there are no issues on iphone.


https://github.com/user-attachments/assets/77cd9cb5-3669-43a8-b802-c10f84fbc1c4

Comment (alexjba) [reply ui/app/AppLayouts/Onboarding/pages/LoginScreen.qml]: it would work on iPhone. But not on all Android phones. It appears there's a race condition in Qt. Enabling the NFC too early in the app lifetime will make it unusable until the user will reset the qt activity by placing the app in the background and back to foreground.

The `Scan keycard` is a response to these two issues found here https://www.figma.com/design/fIuHmhuy8AbjE2nZsCg7Z4/Keycard?node-id=0-1&t=i6FeaNvZRXFaMjqp-1

1. Ensure the NFC can always be enabled
2. If the user will dismiss the nfc drawer we'll be able to trigger it again since it's a user-driven flow.

File: ui/app/AppLayouts/Onboarding/OnboardingFlow.qml [hunk 2]
```diff
@@ -106,11 +106,6 @@ OnboardingStackView {
         property int flow
         property LoginScreen loginScreen: null
 
-        readonly property int loginAccountsModelCount: loginAccountsModel.ModelCount.count
-        onLoginAccountsModelCountChanged: {
-            // NB: have to delay showing the LoginScreen as the model is populated in an async way; therefore can't use `StackView::initialItem`
-            root.restart()
-        }
```
Comment (caybro) [inline ui/app/AppLayouts/Onboarding/OnboardingFlow.qml:113]: Why is this chunk removed?

File: ui/app/AppLayouts/Onboarding/pages/LoginScreen.qml [hunk 3]
```diff
@@ -149,9 +149,6 @@ OnboardingPage {
         })
     }
 
-    Component.onCompleted: {
-        d.setSelectedLoginUser()
-    }
```
Comment (caybro) [inline ui/app/AppLayouts/Onboarding/pages/LoginScreen.qml:154]: Why removed?

File: ui/app/AppLayouts/Onboarding/OnboardingFlow.qml (hunk 2 above)
Comment (saledjenic) [reply ui/app/AppLayouts/Onboarding/OnboardingFlow.qml:113]: Cause it's not needed, otherwise the `restart` is called twice. Why do you think it is needed?

File: ui/app/AppLayouts/Onboarding/pages/LoginScreen.qml (hunk 3 above)
Comment (saledjenic) [reply ui/app/AppLayouts/Onboarding/pages/LoginScreen.qml:154]: Also not needed, cause in `loginUserSelector` we don't listen on `onSelectedProfileKeyIdChanged` which was pointed to the `value` of `ModelEntry` in `LoginUserSelector`, but on `onKeycardCreatedAccountChanged`, which is triggered when the things are ready.

File: ui/app/AppLayouts/Onboarding/OnboardingFlow.qml (hunk 2 above)
Comment (caybro) [reply ui/app/AppLayouts/Onboarding/OnboardingFlow.qml:113]: I remember there being a problem with the model population (and the combo being empty), as the comment suggests. Where is the second place this is being called from?

Comment (saledjenic) [reply ui/app/AppLayouts/Onboarding/OnboardingFlow.qml:113]: From the `OnboardingLayout.qml`, `Component.onCompleted: restartFlow()`.

Comment (saledjenic) [reply ui/app/AppLayouts/Onboarding/OnboardingFlow.qml:113]: Also as I see based on the code here https://github.com/status-im/status-app/blob/master/src/app/modules/onboarding/module.nim#L92-L114  neither `load` nor `let openedAccounts = self.controller.getOpenedAccounts()` are an async calls, so login accounts model is ready always ready before any qml loads.

Comment (saledjenic) [reply ui/app/AppLayouts/Onboarding/OnboardingFlow.qml:113]: I tested that and should not be any issues.

File: ui/app/AppLayouts/Onboarding/pages/LoginScreen.qml (hunk 3 above)
Comment (caybro) [reply ui/app/AppLayouts/Onboarding/pages/LoginScreen.qml:154]: This would be only for the keycard profiles, what about the other ones (password based?)

File: ui/app/AppLayouts/Onboarding/OnboardingFlow.qml (hunk 2 above)
Comment (caybro) [reply ui/app/AppLayouts/Onboarding/OnboardingFlow.qml:113]: Well the 20 failing QML tests tell me otherwise :)

```
[2026-02-04T16:09:27.408Z] 1: Totals: 820 passed, 20 failed, 20 skipped, 0 blacklisted, 147216ms
[2026-02-04T16:09:27.408Z] 1: ********* Finished testing of QmlTests *********
```

Comment (saledjenic) [reply ui/app/AppLayouts/Onboarding/OnboardingFlow.qml:113]: I saw that, will check, fix them.

File: ui/app/AppLayouts/Onboarding/pages/LoginScreen.qml (hunk 3 above)
Comment (saledjenic) [reply ui/app/AppLayouts/Onboarding/pages/LoginScreen.qml:154]: Yes, works fine for both.

**Comment by saledjenic** [comment]
A final... based on discussion here https://discord.com/channels/1210237582470807632/1452689186963329208/1469298323583402116 we've decided to simplify the login flow in a way that we don't read and inspect the keycard state on the app start, but instead display the pin input box always and let the user enter the pin, after that we either display a relevant message/error or log in the user.

**Comment by saledjenic** [comment]
This PR brings several things:
- Fixes broken keycard login on mobile devices (also simplifies login process for the keycard users, a single keycard tap is enough to log in),
- Fixes recovery via seed phrase (after the keycard is locked after missing the pin 3 times, a single keycard tab for the provided seed phrase is enough to recover the keycard and log in)
- Unblocking with puk and unblocking with seed phrase from some onboarding paths are still broken and those buttons are disabled, displaying a tooltop message "Temporarily unavailable, will be available in the next release.". The same states for broken "Create replacement Keycard using the same recovery phrase" and "Start using this profile without Keycard" flows.
- A few more broken things are solved on the keycard lib level, PR https://github.com/status-im/status-keycard-qt/pull/6
- Some checks are done pretty late, like if the user is going to recover, providing a seed prhrase we basically let him do that using any seed phrase (don't check if the provided seed phrase matches the selected profile), and at the end of that process, when tries to log in, the app displays an error, the same/similar
- There are some unhandled things (some edge cases), basically all keycard flows have to be revisited and tested, ideally supported by the tests
- Unhandled scenarios of physical removal or interuption the flow, the app may remain in a blocked state and the only thing the user can do is to restart it (if that happens while migrating profile to a keycard, for example, the local state will be invalid and the user will need to remove the local folder before being able to use to use the same profile).

A single keycar tap login video:

https://github.com/user-attachments/assets/d3a508c4-78ed-4351-b498-ec15c38cf93c

**Review by alaibe — APPROVED** [review]
